Microsoft Edge 151 web platform release notes (Jul. 2026) (#3844)
MicrosoftDocs/edge-developer: microsoft-edge/web-platform/release-notes/151.md

<!-- ====================================================================== -->
## Origin trials

The following are origin trials for new experimental APIs that are available in Microsoft Edge.

Origin trials let you try experimental APIs on your own live website for a limited time.  To learn more about origin trials, see [Use origin trials in Microsoft Edge](../../origin-trials/index.md).

For the full list of available origin trials, see [Microsoft Edge Origin Trials](https://developer.microsoft.com/microsoft-edge/origin-trials/).

| Name | Description | Register |
|---|---|---|
| Soft navigation heuristics | Exposes soft-navigation heuristics for collecting performance metrics in single-page apps. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |
| WebNN [Closed] | [This trial is no longer accepting new registrations.] Build and execute hardware-accelerated machine-learning models directly in your web app. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |

> | focusgroup | todo: condense: Provides focus navigation among a set of focusable elements with the arrow keys. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/6d4b86e7-230c-4d23-ab2b-ee8d05f18702) |

Author: ```suggestion
| focusgroup | todo: condense: Provides focus navigation among a set of focusable elements by using the arrow keys. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/6d4b86e7-230c-4d23-ab2b-ee8d05f18702) |
```

---
title: Microsoft Edge 151 web platform release notes (Jul. 2026)
description: Microsoft Edge 151 web platform release notes (Jul. 2026)
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: article
ms.service: microsoft-edge
ms.date: 07/30/2026
---
# Microsoft Edge 151 web platform release notes (Jul. 2026)

The following are the new web platform features and updates in Microsoft Edge 151, which releases on Jul. 30, 2026.

To stay up-to-date and get the latest web platform features, download a preview channel of Microsoft Edge (Beta, Dev, or Canary); go to [Become a Microsoft Edge Insider](https://www.microsoft.com/edge/download/insider).

**Detailed contents:**

* [Microsoft Edge DevTools](#microsoft-edge-devtools)
* [WebView2](#webview2)
* [CSS features](#css-features)
   * [CSS `ruby-overhang` property](#css-ruby-overhang-property)
   * [Changed the `position-anchor` initial value to `normal`](#changed-the-position-anchor-initial-value-to-normal)
   * [CSS `alpha()` function](#css-alpha-function)

>    * [`animation-trigger` actions no long auto-rewind](#animation-trigger-actions-no-long-auto-rewind)

Reviewer: ```suggestion
   * [`animation-trigger` actions no longer auto-rewind](#animation-trigger-actions-no-longer-auto-rewind)
```

* [HTML features](#html-features)
   * [`<usermedia>` element](#usermedia-element)
   * [Reference target](#reference-target)
   * [`aria-actions` attribute](#aria-actions-attribute)
   * [`shadowrootslotassignment` attribute for declarative shadow DOM](#shadowrootslotassignment-attribute-for-declarative-shadow-dom)
* [Web APIs](#web-apis)
   * [Device event permission request API](#device-event-permission-request-api)
   * [Duplicate successive navigations are ignored](#duplicate-successive-navigations-are-ignored)
   * [Cross-origin redirect timing in the Navigation Timing API](#cross-origin-redirect-timing-in-the-navigation-timing-api)
   * [LanguageDetector support for Traditional and Simplified Chinese](#languagedetector-support-for-traditional-and-simplified-chinese)
   * [`soft-navigation` and `interaction-contentful-paint` performance entries](#soft-navigation-and-interaction-contentful-paint-performance-entries)
   * [Exposed the `FontFaceSet` interface globally](#exposed-the-fontfaceset-interface-globally)
   * [Speculation rules: `form_submission` field](#speculation-rules-form_submission-field)

>    * [Support for unspoken punctuation in the Web Speech API](#support-for-unspoken-punctuation-in-the-web-speech-api)

Reviewer: ```suggestion
```

<!-- ------------------------------ -->
#### Changed the `position-anchor` initial value to `normal`

The initial value for the `position-anchor` CSS property is now `normal`.  Previously, the initial value was `none`.

This aligns Microsoft Edge with other browsers and with the anchor positioning specification.

* If `position-area` is set to `none`, `normal` behaves like `none`.
* If `position-area` is not set to `none`, `normal` behaves like `auto`.

See also:
* [`position-anchor` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/position-anchor) at MDN.
* [`position-area` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/position-area) at MDN.

<!-- ------------------------------ -->
#### CSS `alpha()` function

The CSS `alpha()` function creates a new color by applying a specified alpha value to an existing color.

For example, `alpha(from red / 0.5)` creates a semi-transparent red color with 50% opacity.

See also:
* [Relative Alpha Colors](https://drafts.csswg.org/css-color-5/#relative-alpha) in _CSS Color Module Level 5_.

<!-- ------------------------------ -->

> #### `animation-trigger` actions no long auto-rewind

Reviewer: ```suggestion
#### `animation-trigger` actions no longer auto-rewind
```

#### `aria-actions` attribute

The `aria-actions` HTML attribute exposes secondary actions for composite interactive widgets.

For example, you can use the `aria-actions` attribute to expose the close button of a tab, which is a secondary action for the tab widget, as follows:

```html
<div class="tab" aria-actions="close-tab">
  Tab title
  <button id="close-tab">Close</button>
</div>
```

Secondary actions placed within composite interactive widgets is a common UI pattern.  The `aria-actions` attribute allows you to expose these secondary action buttons directly, to improve discoverability for assistive technology (AT) users.

See also:
* [Feature: `aria-actions`](https://chromestatus.com/feature/5161589307867136) at Chrome Platform Status.

<!-- ------------------------------ -->
#### `shadowrootslotassignment` attribute for declarative shadow DOM

The `shadowrootslotassignment` attribute of the `<template>` element allows declarative shadow roots to use manual slot assignment.

Previously, manual slot assignment was only available via JavaScript, by using `attachShadow({slotAssignment: "manual"})`.

The `shadowrootslotassignment` attribute takes the following values:

> * `named` - The default value, which preserves the current behavior.  todo: state the current behavior explicitly/directly

Author: state the current behavior explicitly/directly

> * `manual` - Causes declarative shadow roots to use manual slot assignment.

Reviewer: ````suggestion
You can now declare manual slot assignment from HTML:

```html
<template shadowrootslotassignment="manual">
</template>
```
````

This is useful for search forms where the result URL can be predicted in advance.  For example, to prerender the result of a search form that navigates to `/search?q=foo`, use the following speculation rules:

```html
<script type="speculationrules">
{
  "prerender": [
    {
      "urls": ["/search?q=foo"],
      "form_submission": true
    }
  ]
}
</script>
```

See also:
* [Speculation Rules API](https://developer.mozilla.org/docs/Web/API/Speculation_Rules_API) at MDN.

<!-- ------------------------------ -->
#### Support for unspoken punctuation in the Web Speech API

The `SpeechRecognition` interface of the Web Speech API now has an `unspokenPunctuation` boolean attribute.

When set to `true`, the `unspokenPunctuation` attribute directs the speech-recognition engine to automatically infer and insert punctuation marks (such as periods, commas, and question marks) based on the user's pauses, grammatical structure, and intonation, without requiring explicit spoken punctuation commands.

See also:

> * [Convert speech to text with the SpeechRecognition API](../speech-recognition-api.md)

Reviewer: ```suggestion
<!-- ------------------------------ todo: the unspokenPunctuation is not supported in Edge yet. Uncomment and move to the right release notes when support is added.
#### Support for unspoken punctuation in the Web Speech API

The `SpeechRecognition` interface of the Web Speech API now has an `unspokenPunctuation` boolean attribute.

When set to `true`, the `unspokenPunctuation` attribute directs the speech-recognition engine to automatically infer and insert punctuation marks (such as periods, commas, and question marks) based on the user's pauses, grammatical structure, and intonation, without requiring explicit spoken punctuation commands.

See also:
* [Convert speech to text with the SpeechRecognition API](../speech-recognition-api.md)
-->
```

| HTML in canvas | Enables rendering HTML in canvas with new drawing methods and the `paint` event. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/a297467e-0030-4c4c-8739-48e130026c03) |
| Gamepad raw input event | Instead of relying on frequent polling via `navigator.getGamepads()`, web pages can listen for `gamepadrawinputchanged` events.  These events are fired whenever changes are detected between input frames from the same device. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/4ecdb530-835b-447d-b27e-0b3be8e0412b) |
| Autofill Event | Detects when browser autofill updates form controls, for adapting custom UI and validation. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/43534059-5696-4eaf-8bbd-eba4c4f86e88) |
| WebAssembly Custom Descriptors V2 | Stores data associated with source-level types more efficiently, in custom descriptor objects. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/9c75e7f3-49cb-43de-9745-7458ff9c3689) |

> | Speculative Load measurement | todo: condense: `performance.speculations` will provide information on pending and used preloads, as well as on speculation rules&#39; prefetches and prerenders.  That will enable developers to know what speculative loads are happening on their sites, and be able to estimate their cost. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/246de200-45d2-4f44-8047-001d385c90c5) |

Author: condense OT desc: Speculative Load measurement

Reviewer: ```suggestion
| Speculative Load measurement | `performance.speculations` exposes data about used and unused preloads, prefetches, and prerenders, helping you measure the effectiveness of speculative loading strategies and optimize performance-versus-cost tradeoffs. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/246de200-45d2-4f44-8047-001d385c90c5) |
```

| WebMCP | Enables your site to register tools for an in-browser agent to complete tasks on behalf of a user. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/0b76fe60-b266-458e-a285-04e375c0c31a) |

> | Email Verification Protocol | todo: condense: Get automatically verified email addresses through browser autofill. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/683027d2-a798-460f-9b35-03a95002d9af) |

Author: condense OT desc: Email Verification Protocol

Reviewer: ```suggestion
| Email Verification Protocol | Enables websites to verify email ownership using cryptographic tokens issued by email providers, reducing reliance on email OTPs and streamlining sign-up, sign-in, and recovery flows. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/683027d2-a798-460f-9b35-03a95002d9af) |
```

These quick succession navigations are often caused by accidental double-clicks, and can waste resources by canceling the ongoing navigation and starting a new one.

This optimization improves performance and the user experience.

See also:
* [Experiment with keeping older navigation on duplicate navigation](https://issues.chromium.org/issues/366060351) in the Chromium bug tracker.

<!-- ------------------------------ -->
#### Cross-origin redirect timing in the Navigation Timing API

The `Timing-Allow-Origin` header, when used in chains of cross-origin redirects, provides values for `redirectCount`, `redirectStart` and `redirectEnd` to the destination document, via the Navigation Timing API.

This lets you measure and optimize your cross-origin redirects.

See also:
* [Navigation timing](https://developer.mozilla.org/docs/Web/API/Performance_API/Navigation_timing) at MDN.
* [Timing-Allow-Origin header](https://developer.mozilla.org/docs/Web/HTTP/Reference/Headers/Timing-Allow-Origin) at MDN.

<!-- ------------------------------ -->

> #### LanguageDetector support for Traditional and Simplified Chinese<!-- todo: wait for team to confirm this is available in edge -->

Author: team: confirm "LanguageDetector support for Traditional and Simplified Chinese" is available in edge

Reviewer: ```suggestion
#### LanguageDetector support for Traditional and Simplified Chinese
```

See also:
* [Convert speech to text with the SpeechRecognition API](../speech-recognition-api.md)

<!-- ------------------------------ -->
#### Wheel event momentum

The `WheelEvent` interface now includes a `momentum` attribute, which indicates whether the event is part of a momentum scroll.

The `momentum` attribute is useful, for example, on devices with trackpads, where a user can perform a fling gesture to scroll content. Many devices continue to fire wheel events for a while, to simulate scrolling inertia.

The `momentum` attribute differentiates those simulated events from the events that are fired during the physical trackpad interaction.

See also:
* [WheelEvent](https://developer.mozilla.org/docs/Web/API/WheelEvent) at MDN.

<!-- ------------------------------ -->
#### XML parsing in Rust for non-XSLT scenarios

Microsoft Edge now uses a Rust-based XML parser, in scenarios where XSLT processing isn't required.

The Rust XML parser improves security by eliminating memory-corruption bugs in XML parsing.  This applies to:
* The DOMParser Web API.

> * `responseXML` of `XMLHttpRequest`.  todo: the `responseXML` attribute of the `<XMLHttpRequest>` element?

Author: ```suggestion
* The `responseXML` property on an `XMLHttpRequest` (XHR) object.
```

<!-- ------------------------------ -->
#### Navigation: Ignore duplicate navigations

Prevents an ongoing navigation from being unnecessarily canceled by a new, identical navigation that is initiated in quick succession.

This optimization improves performance and the user experience, by not wasting resources on a duplicate request, which can be caused by accidental double-clicks.

See also:
* []

<!-- ------------------------------ -->
#### No Auto-Rewind for AnimationTrigger Play methods

AnimationTrigger Play methods will not auto-rewind; that is, AnimationTrigger Play methods will not restart a finished animation.

`play`, `play-forwards` and `play-backwards` are values that, when specified in the association between an `AnimationTrigger` and an `Animation`, instruct the trigger to play the animation.

This Chromestatus (todo: what's "Chromestatus"?) feature covers this specific aspect of the behavior of these keywords:
* When the relevant animation has already run to completion and these actions (`play`, `play-forwards`, `play-backwards`) are invoked, they will not cause the animation to restart; the animation will not "auto-rewind".

> todo: bug: change `play-forwards` to `play-forward`

Author: bug/typo in API name.  Can we get "play-forwards" corrected and changed to "play-forward"?

@captainbrosset

Reviewer: No, this ship has sailed. It's been standardized this way.
